### PR TITLE
Migrate Delete operation to the new translation layer

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Description
As for the title, migrate the Top-level delete operations to the new translation layer (when subscription is enabled is still using the previous implementation).
Apart from that, a minor fix QueryAST print method.

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Thanks for the Neo4j GraphQL Toolbox updates.

The Neo4j GraphQL Toolbox has now been torn down - reopening this PR will republish it.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3404 | 3404 | 33 | 43 | 1134 |
|  aggregations.TopLevelAggregateWithMultipleFields | 6802 | 6802 | 40 | 55 | 1134 |
|  aggregations.NestedAggregation | 16544 | 16544 | 62 | 77 | 2174 |
|  aggregations.AggregationWithWhere | 11970 | 11970 | 34 | 58 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116478 | 22116478 | 2109 | 2246 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116478 | 22116478 | 2069 | 2201 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9701 | 9701 | 33 | 47 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9928 | 9928 | 30 | 44 | 2174 |
|  aggregations.DeeplyNestedCount | 11062184 | 11062184 | 2917 | 2898 | 2008534 |
|  aggregations.InterfacesAggregations | 6242 | 6242 | 42 | 64 | 2080 |
|  aggregations.InterfacesAggregationsWithTwoFields | 11444 | 11444 | 65 | 89 | 2080 |
|  batch-create.BatchCreate | 4200 | 4200 | 109 | 167 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 48 | 65 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 143 | 192 | 3003 |
|  connections.Connection | 14049 | 14049 | 97 | 64 | 2174 |
|  connections.NestedConnection | 41068 | 41068 | 105 | 132 | 4516 |
|  create.SimpleMutation | 7 | 7 | 45 | 54 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 24 | 22 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 245 | 203 | 1040 |
|  delete.NestedDeleteInUpdate | 19165 | 19165 | 146 | 251 | 2040 |
|  2871.NestedRelationshipFilter | 23082 | 23083 | 49 | 65 | 4434 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 33 | 44 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22891 | 22891 | 66 | 99 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 33 | 38 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 47 | 107 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 18 | 21 | 1040 |
|  query.SimpleQueryWithRelationship | 16129 | 16129 | 37 | 38 | 2174 |
|  query.QueryWhere | 9691 | 9676 | 31 | 36 | 2166 |
|  query.SimpleQueryWithNestedWhere | 9858 | 9843 | 44 | 88 | 2166 |
|  query.Nested | 10095502 | 10095502 | 6836 | 6561 | 2008534 |
|  query.NestedWithFilter | 10074351 | 10074351 | 6631 | 6727 | 2004000 |
|  query.OrFilterOnRelationships | 43661 | 43007 | 152 | 149 | 2033 |
|  query.OrFilterOnRelationshipsAndNested | 36744 | 35862 | 178 | 245 | 2033 |
|  query.QueryWithNestedIn | 13857 | 13451 | 43 | 74 | 1379 |
|  query.NestedConnectionWhere | 9801 | 9801 | 49 | 69 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9893 | 9859 | 68 | 86 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 18746 | 18758 | 114 | 160 | 1586 |
|  query.NestedWithRelationshipSingleFilters | 3865 | 3865 | 159 | 190 | 1134 |
|  query.Fulltext | 64 | 64 | 27 | 34 | 16 |
|  query.FulltextWithNestedQuery | 558 | 558 | 35 | 47 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 14059 | 14059 | 43 | 46 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 14059 | 14059 | 61 | 68 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 14185 | 14184 | 52 | 59 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 14185 | 14184 | 86 | 94 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 14661 | 14965 | 99 | 118 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 14480 | 14493 | 93 | 106 | 2174 |
|  sorting.SortMultipleTypes | 3498 | 3498 | 74 | 88 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14460 | 14435 | 92 | 102 | 2174 |
|  sorting.SortOnNestedFields | 14049 | 14049 | 54 | 47 | 2174 |
|  sorting.SortDeeplyNestedFields | 43148 | 43148 | 80 | 77 | 4516 |
|  sorting.ConnectionWithSort | 3281 | 3281 | 61 | 68 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 48 | 60 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 47 | 54 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 300 | 245 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 282 | 275 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 91 | 109 | 2002 |

</details>
        
Old Schema Generation: 23.814s
Schema Generation: 23.567s
Old Subgraph Schema Generation: 26.985s
Subgraph Schema Generation: 26.950s

File: packages/graphql/src/translate/queryAST/ast/operations/DeleteOperation.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        for (const nestedDeleteOperation of this.nestedOperations) {
+            const { clauses } = nestedDeleteOperation.transpile(context);
+            nestedOperations.push(...clauses.map((c) => new Cypher.Call(c).innerWith("*")));
+        }
+        return nestedOperations;
+    }
+
+    private validateSelection(selection: SelectionClause): asserts selection is Cypher.Match {
+        if (!(selection instanceof Cypher.Match)) {
+            throw new Error("Yield is not a valid selection for Delete Operation");
```
Comment (mjfwebb) [inline packages/graphql/src/translate/queryAST/ast/operations/DeleteOperation.ts]: Where does the word "yield" come from?

File: packages/graphql/src/translate/translate-delete.ts [hunk 2]
```diff
@@ -19,25 +19,58 @@
...
+    varName,
+}: {
+    context: Neo4jGraphQLTranslationContext;
+    entityAdapter: EntityAdapter;
+    resolveTree: ResolveTree;
+    varName: string;
+}) {
+    const operationsTreeFactory = new QueryASTFactory(context.schemaModel, context.experimental);
+
+    if (!entityAdapter) throw new Error("Entity not found");
```
Comment (mjfwebb) [inline packages/graphql/src/translate/translate-delete.ts]:
```suggestion
    if (!entityAdapter) {
        throw new Error("Entity not found");
    }
```

**Review by mjfwebb — DISMISSED** [review]
Great!

File: packages/graphql/src/translate/queryAST/ast/operations/DeleteOperation.ts (hunk 1 above)
Comment (MacondoExpress) [reply packages/graphql/src/translate/queryAST/ast/operations/DeleteOperation.ts]: Is the Cypher statement Yield part of the SelectionClause, the reason why for this assertion it's just to keep the implementation clean and avoid "if branches" for something we currently don't don't support!

**Review by mjfwebb — APPROVED** [review]

**Pull request merged**